core/src/test/java/org/elasticsearch/index/query/SpanNotQueryBuilderTest.java

```diff
@@ -0,0 +1,202 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.index.query;
+
+import org.apache.lucene.search.Query;
+import org.apache.lucene.search.spans.SpanNotQuery;
+import org.apache.lucene.search.spans.SpanQuery;
+import org.elasticsearch.common.xcontent.XContentBuilder;
+import org.elasticsearch.common.xcontent.XContentFactory;
+import org.elasticsearch.common.xcontent.XContentParser;
+import org.junit.Test;
+
+import java.io.IOException;
+
+import static org.elasticsearch.index.query.QueryBuilders.spanNearQuery;
+import static org.elasticsearch.index.query.QueryBuilders.spanTermQuery;
+import static org.hamcrest.Matchers.containsString;
+import static org.hamcrest.Matchers.equalTo;
+
+public class SpanNotQueryBuilderTest extends BaseQueryTestCase<SpanNotQueryBuilder> {
+
+    @Override
+    protected Query doCreateExpectedQuery(SpanNotQueryBuilder testQueryBuilder, QueryParseContext context) throws IOException {
+        SpanQuery include = (SpanQuery) testQueryBuilder.include().toQuery(context);
+        SpanQuery exclude = (SpanQuery) testQueryBuilder.exclude().toQuery(context);
+        return new SpanNotQuery(include, exclude, testQueryBuilder.pre(), testQueryBuilder.post());
+    }
+
+    @Override
+    protected SpanNotQueryBuilder doCreateTestQueryBuilder() {
+        SpanTermQueryBuilder includeQuery = new SpanTermQueryBuilderTest().createTestQueryBuilder();
+        // we need same field name and value type as bigQuery for little query
+        String fieldName = includeQuery.fieldName();
+        Object value;
+        switch (fieldName) {
+            case BOOLEAN_FIELD_NAME: value = randomBoolean(); break;
+            case INT_FIELD_NAME: value = randomInt(); break;
+            case DOUBLE_FIELD_NAME: value = randomDouble(); break;
+            case STRING_FIELD_NAME: value = randomAsciiOfLengthBetween(1, 10); break;
+            default : value = randomAsciiOfLengthBetween(1, 10);
```

Comment:
I think if we get in that other PR I just reviewd we can reuse here the new method that you introduced there? :)